ibis/spark/client.py

```
@@ -0,0 +1,295 @@
+
+    def list_tables(self, like=None, database=None):
+        """
+        List tables in the current (or indicated) database. Like the SHOW
+        TABLES command.
+
+        Parameters
+        ----------
+        like : string, default None
+          e.g. 'foo*' to match all tables starting with 'foo'
+        database : string, default None
+          If not passed, uses the current/default database
+
+        Returns
+        -------
+        results : list of strings
+        """
+        results = [t.name for t in self._catalog.listTables(dbName=database)]
+        if like:
+            results = [
+                database_name
+                for database_name in results
+                if re.match(like, database_name) is not None
+            ]
+
+        return results
+
+    def set_database(self, name):
+        """
+        Set the default database scope for client
+        """
+        self._catalog.setCurrentDatabase(name)
+
+    def exists_database(self, name):
+        """
+        Checks if a given database exists
+
+        Parameters
+        ----------
+        name : string
+          Database name
+
+        Returns
+        -------
+        if_exists : boolean
+        """
+        return bool(self.list_databases(like=name))
+
+    def list_databases(self, like=None):
+        """
+        List databases in the Spark SQL cluster.
+
+        Parameters
+        ----------
+        like : string, default None
+          e.g. 'foo*' to match all tables starting with 'foo'
+
+        Returns
+        -------
+        results : list of strings
+        """
+        results = [db.name for db in self._catalog.listDatabases()]
+        if like:
+            results = [
+                database_name
+                for database_name in results
+                if re.match(like, database_name) is not None
+            ]
+
+        return results
+
+    def get_schema(self, table_name, database=None):
```

If you're not going to handle a non-`None` value for `database` you should raise an exception indicating as such.